Source/WebCore/platform/mediastream/mac/CoreAudioSharedUnit.cpp

```diff
@@ -681,8 +682,24 @@ bool CoreAudioSharedUnit::migrateToNewDefaultDevice(const CaptureDevice& capture
     handleNewCurrentMicrophoneDevice(WTFMove(*device));
     return true;
 }
-#endif
 
+void CoreAudioSharedUnit::prewarmAudioUnitCreation(CompletionHandler<void()>&& callback)
+{
+    if (!m_audioUnitCreationWarmupPromise) {
+        m_audioUnitCreationWarmupPromise = invokeAsync(WorkQueue::create("CoreAudioSharedUnit AudioUnit creation").get(), [weakThis = WeakPtr { *this }] () mutable {
+            callOnMainRunLoop([weakThis = WTFMove(weakThis), vpioUnitOrError = createAudioUnit(true)] () mutable {
+                if (weakThis && vpioUnitOrError.has_value())
+                    weakThis->setStoredVPIOUnit(WTFMove(vpioUnitOrError.value()));
+            });
+            return WTF::GenericNonExclusivePromise::createAndResolve();
+        });
+    }
+
+    m_audioUnitCreationWarmupPromise->whenSettled(RunLoop::main(), [callback = WTFMove(callback)] (auto&&) mutable {
```

Comment:
`auto&&` can be removed; the lambda given to whenSettled takes either an Expected or void